ocw/lib/gke.py

```diff
@@ -0,0 +1,77 @@
+import os
+import subprocess
+from datetime import datetime, timezone
+import kubernetes
+import googleapiclient.discovery
+from google.oauth2 import service_account
+from ocw.lib.gce import GCE
+
+
+class GKE(GCE):
+    __instances = {}
+
+    def __new__(cls, vault_namespace):
+        if vault_namespace not in GKE.__instances:
+            GKE.__instances[vault_namespace] = self = object.__new__(cls)
+            self.__gke_client = None
+            self.__kubectl_client = {}
+            self.init()
+        return GKE.__instances[vault_namespace]
+
+    def get_project_id(self):
+        return self.get_data()["project_id"]
+
+    def gke_client(self):
+        if self.__gke_client is None:
+            self.private_key_data = self.get_data()
+            credentials = service_account.Credentials.from_service_account_info(self.private_key_data)
+            self.__gke_client = googleapiclient.discovery.build('container', 'v1', credentials=credentials)
+
+        return self.__gke_client
+
+    def kubectl_client(self, zone, cluster):
+        cluster_name = cluster["name"]
+        kube_dir = "/root/.kube"
+        kubeconfig = f"{kube_dir}/gke_config_{zone}_{cluster_name}"
+        try:
+            self.cmd_exec(f"gcloud auth login  --project={self.get_project_id()} --cred-file=/var/pcw/qac/GKE.json --quiet")
+            self.cmd_exec(f"gcloud container clusters get-credentials {cluster_name} " +
+                          f"--zone {zone} --project {self.get_project_id()}", {"KUBECONFIG": kubeconfig})
+            if not os.path.exists(kubeconfig):
+                raise FileNotFoundError("{kubeconfig} doesn't exists")
+        except subprocess.CalledProcessError as exception:
+            self.log_err(f"Cannot get the kubeconfig for the cluster {cluster_name} on region {zone}, reason: {str(exception)}")
+            return None
```

Comment:
```suggestion
            raise
```
no point to return None it will fail anyway just throw further whatever you have caught here

Reply:
but I remove the None I have a lint error
`ocw/lib/gke.py:29:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)`

Maybe the best solution could be remove the try and allowing that when an error raises without caching. I did that